File: c/euler1_1.c
// Euler1 in C

#include <stdio.h>

int euler1(int n, int acc) {
	return n == 0 ? acc :
		euler1( n-1, acc+(n%3==0 || n%5==0 ? n : 0) );
}

int main() {
	printf("euler1 = %i\n", euler1(999, 0));
	return 0;
}

File: lua/euler1_1.lua
#!/usr/bin/lua5.4
-- Euler1 in Lua

function euler(n, acc)
    if n == 1 then
        return acc
  elseif n%3 == 0 or n%5 == 0 then
        return euler(n-1, acc+n)
  else
      return euler(n-1, acc)
  end
end

function euler1(n)
    return euler(n, 0)
end

print (string.format("Euler1 = %d", euler1(999)))
